src/module.c

```diff
@@ -50,7 +50,7 @@ static int parseLabelsFromArgs(RedisModuleString **argv, int argc, size_t *label
         	RedisModule_StringPtrLen(key, &keyLen);
         	RedisModule_StringPtrLen(value, &valueLen);
         	if(keyLen==0 || valueLen==0 || strpbrk( RedisModule_StringPtrLen(value, NULL), "(),")){
-        		free(labelsResult);
+        		FreeLabels(labelsResult, i); // need to release prior key values too
         		return REDISMODULE_ERR;
         	}
 
```

Comment:
@gkorland why not use the context here and depend on automemory?

Reply:
anyway, its not crtical for now, but we might want to consider all the parse function to use ctx to allow the automemory to clean on failure.